client/main.lua

```diff
@@ -24,193 +23,219 @@ local currentGear = {
+    end
+
+    local radiusBlip = AddBlipForRadius(Config.CoralLocations[currentDivingLocation.area].coords.area, 100.0)
+
     SetBlipRotation(radiusBlip, 0)
     SetBlipColour(radiusBlip, 47)
+
     currentDivingLocation.blip.radius = radiusBlip
-    local labelBlip = AddBlipForCoord(Config.CoralLocations[currentDivingLocation.area].coords.Area)
+
+    local labelBlip = AddBlipForCoord(Config.CoralLocations[currentDivingLocation.area].coords.area)
+
     SetBlipSprite(labelBlip, 597)
     SetBlipDisplay(labelBlip, 4)
     SetBlipScale(labelBlip, 0.7)
     SetBlipColour(labelBlip, 0)
     SetBlipAsShortRange(labelBlip, true)
+
     BeginTextCommandSetBlipName('STRING')
     AddTextComponentSubstringPlayerName(Lang:t("info.diving_area"))
     EndTextCommandSetBlipName(labelBlip)
+
     currentDivingLocation.blip.label = labelBlip
-    for k, v in pairs(Config.CoralLocations[currentDivingLocation.area].coords.Coral) do
+
+    for k, v in pairs(Config.CoralLocations[currentDivingLocation.area].coords.coral) do
         if Config.UseTarget then
-            exports['qb-target']:AddBoxZone('diving_coral_zone_'..k, v.coords, v.length, v.width, {
-                name = 'diving_coral_zone_'..k,
-                heading = v.heading,
-                debugPoly = false,
-                minZ = v.coords.z - 3,
-                maxZ = v.coords.z + 2
-            }, {
+            zones[k] = exports.ox_target:addBoxZone({
+                coords = v.coords,
+                size = v.size,
+                rotation = v.rotation,
                 options = {
                     {
-                        label = Lang:t("info.collect_coral"),
+                        name = 'qb-diving:coral-' .. k,
                         icon = 'fa-solid fa-water',
-                        action = function()
+                        label = Lang:t("info.collect_coral"),
+                        distance = 2.0,
+                        onSelect = function(_)
                             takeCoral(k)
                         end
                     }
-                },
-                distance = 2.0
+                }
             })
         else
-            zones[k] = BoxZone:Create(v.coords, v.length, v.width, {
-                name = 'diving_coral_zone_'..k,
-                heading = v.heading,
-                debugPoly = false,
-                minZ = v.coords.z - 3,
-                maxZ = v.coords.z + 2
-            })
-            zones[k]:onPlayerInOut(function(inside)
-                if inside then
+            zones[k] = lib.zones.box({
+                coords = v.coords,
+                size = v.size,
+                rotation = v.rotation,
+                onEnter = function(_)
                     currentArea = k
-                    exports['qb-core']:DrawText(Lang:t("info.collect_coral_dt"))
-                else
+
+                    lib.showTextUI(Lang:t("info.collect_coral_dt"))
+                end,
+                onExit = function(_)
```

Comment:
```suggestion
                onExit = function()
```
Remove the pointless variable.